network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/ui/NetpVpnAccessRevokedDialogMonitor.kt

```diff
@@ -0,0 +1,78 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.subscription.ui
+
+import android.app.Activity
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.browser.api.ActivityLifecycleCallbacks
+import com.duckduckgo.common.ui.view.dialog.TextAlertDialogBuilder
+import com.duckduckgo.common.ui.view.dialog.TextAlertDialogBuilder.EventListener
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.navigation.api.GlobalActivityStarter
+import com.duckduckgo.networkprotection.impl.store.NetworkProtectionRepository
+import com.duckduckgo.networkprotection.subscription.R.string
+import com.duckduckgo.subscriptions.api.SubscriptionScreens.SubscriptionScreenNoParams
+import com.squareup.anvil.annotations.ContributesMultibinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@ContributesMultibinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class NetpVpnAccessRevokedDialogMonitor @Inject constructor(
+    private val networkProtectionRepository: NetworkProtectionRepository,
+    private val globalActivityStarter: GlobalActivityStarter,
+    @AppCoroutineScope private val coroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+) : ActivityLifecycleCallbacks {
+    override fun onActivityResumed(activity: Activity) {
+        super.onActivityResumed(activity)
+        coroutineScope.launch(dispatcherProvider.io()) {
+            if (networkProtectionRepository.vpnAccessRevoked) {
+                withContext(dispatcherProvider.main()) {
+                    TextAlertDialogBuilder(activity)
+                        .setTitle(string.netpDialogVpnAccessRevokedTitle)
+                        .setMessage(string.netpDialogVpnAccessRevokedBody)
+                        .setPositiveButton(string.netpDialogVpnAccessRevokedPositiveAction)
+                        .setNegativeButton(string.netpDialogVpnAccessRevokedNegativeAction)
+                        .addEventListener(
+                            object : EventListener() {
+                                override fun onPositiveButtonClicked() {
+                                    globalActivityStarter.start(activity, SubscriptionScreenNoParams)
+                                    resetVpnAccessRevokedState()
+                                }
+
+                                override fun onNegativeButtonClicked() {
+                                    resetVpnAccessRevokedState()
+                                }
+                            },
+                        )
+                        .show()
+                }
+            }
+        }
+    }
+
+    private fun resetVpnAccessRevokedState() {
```

Comment:
should this rather be a `suspend fun` and add `withContext`

Reply:
Won’t fix